File: 1534_count_good_triplets.cpp
class Solution {
public:
    int countGoodTriplets(vector<int>& z, int x, int y, int w) {
        int s = 0, n = z.size();
        for(int i = 0; i < n; ++i)
            for(int j = i+1; j < n; ++j)
                if(abs(z[i]-z[j]) <= x)
                    for(int k = j+1; k < n; ++k)
                        s += abs(z[j]-z[k]) <= y && abs(z[i]-z[k]) <= w;
        return s;
    }
};

File: 1534_count_good_triplets.py
class Solution(object):
    def countGoodTriplets(self, z, x, y, w):
        r, n = 0, len(z)
        for i in range(n):
            for j in range(i + 1, n):
                if abs(z[i] - z[j]) <= x:
                    for k in range(j + 1, n):
                        if abs(z[j] - z[k]) <= y and abs(z[i] - z[k]) <= w:
                            r += 1
        return r
